### PR TITLE
Optionally disable consider-using-join for non-empty separator

### LINKED ISSUE #8701: Disable consider-using-join for non-empty separator
### Current problem

In 2.16.0, consider-using-join was extended to include non-empty separators. This makes pylint complains about the following code:

```py
def test(values: List[str]):
    data = ""
    for value in values:
        data += f"prefix{value}suffix"
    return data
```

However I don't think the suggested refactor is an improvement: prefix and suffix are duplicated and the empty list case need to be handled specifically

```py
def test(values: List[str]):
    if not values:
        return ""
    return "prefix" + "suffixprefix".join(values) + "suffix"
```

The current options are:
 - disable the rule completely. This has the drawback of loosing the previous rule for empty separators.
 - use a "pylint: disable" comment. However having to use this on regular basis creates bad habits and reduce the value of linting

### Desired solution

Would it be possible to have an option to disable the rule only for the non-empty separator case?

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by theirix**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |
| ✓   | :scroll: Docs          |

## Description

Skip ``consider-using-join`` check for non-empty separators unless a ``suggest-join-with-non-empty-separator`` option is set to ``yes``.

Closes #8701

### CODE REVIEW

File: tests/functional/c/consider/consider_join2.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# pylint: disable=missing-docstring,invalid-name,undefined-variable,multiple-statements
```
Comment (Pierre-Sassoulas) [inline tests/functional/c/consider/consider_join2.py:1]: I would name this file ``consider_join_for_non_empty_separator.py`` (can't suggest to rename file during review).

File: pylint/checkers/refactoring/refactoring_checker.py [hunk 2]
```diff
@@ -506,6 +506,18 @@ class RefactoringChecker(checkers.BaseTokenChecker):
...
         ),
+        (
+            "suggest-join-with-non-empty-separator",
+            {
+                "default": False,
+                "type": "yn",
+                "metavar": "<y or n>",
+                "help": (
+                    "Suggest using join for consider-using-join "
+                    "when a non-empty item separator is used."
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/refactoring/refactoring_checker.py:517]:
```suggestion
                    "Let 'consider-using-join' be raised when the separator to "
                    "join on would be non-empty (resulting in expected fixes "
                    "of the type: ``"- " + "\n- ".join(items)``)"
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Looks pretty good already 👍

File: tests/functional/c/consider/consider_join2.py (hunk 1 above)
Comment (theirix) [reply tests/functional/c/consider/consider_join2.py]: Thank you, make sense. Renamed in an additional commit.

File: pylint/checkers/refactoring/refactoring_checker.py [hunk 3]
```diff
@@ -506,6 +506,19 @@ class RefactoringChecker(checkers.BaseTokenChecker):
                 "and no message will be printed.",
             },
         ),
+        (
+            "suggest-join-with-non-empty-separator",
+            {
+                "default": False,
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/refactoring/refactoring_checker.py:512]:
```suggestion
                "default": True,
```
I guess the default should stay the same than before or it's a breaking change in a minor. Waiting for another maintainer opinion. (Documentation in fragment would also be different if we change that but let's not change everything yet).

**Review by Pierre-Sassoulas — APPROVED** [review]
Great MR, thank you ! (Will be released in 3.1.0)

**Pull request merged**